src/sql/workbench/contrib/notebook/browser/calloutDialog/imageCalloutDialog.ts

```diff
@@ -196,10 +199,14 @@ export class ImageCalloutDialog extends Modal {
 
 	public insert(): void {
 		this.hide('ok');
+		let imgPath = this._imageUrlInputBox.value;
+		let imageName = path.basename(imgPath);
 		this._selectionComplete.resolve({
-			insertEscapedMarkdown: `![](${escapeUrl(this._imageUrlInputBox.value)})`,
-			imagePath: this._imageUrlInputBox.value,
-			embedImage: this._imageEmbedCheckbox.checked
+			embedImage: this._imageEmbedCheckbox.checked,
+			// check for spaces and remove them in imageName.
+			// if spaces in image path replace with &#32; as per https://github.com/microsoft/vscode/issues/11933#issuecomment-249987377
+			insertEscapedMarkdown: this._imageEmbedCheckbox.checked ? `![${imageName}](attachment:${imageName.replace(' ', '')})` : `![](${imgPath.replace(' ', '&#32;')})`,
```

Comment:
This will only replace the first instance of a space in the filename. Instead you can use 
`imgPath.replace(/\s/g, '&#32;')`